### PR TITLE
SED-2113 Support streaming for psql queries

### PR DESCRIPTION
**Pull request by david-stephan**

### CODE REVIEW

File: step-framework-parent/step-framework-collections-mongodb/src/main/java/step/core/collections/mongodb/MongoDBCollection.java [hunk 1]
```diff
@@ -109,6 +109,11 @@ public Stream<T> find(Filter filter, SearchOrder order, Integer skip, Integer li
 		return find(filter, order, skip, limit, maxTime, null);
 	}
 
+	@Override
+	public Stream<T> findCloseableStream(Filter filter, SearchOrder order, Integer skip, Integer limit, int maxTime) {
```
Comment (jeromecomte) [inline step-framework-parent/step-framework-collections-mongodb/src/main/java/step/core/collections/mongodb/MongoDBCollection.java:113]: Not sure of the naming. findWithStreaming or something stating the main difference with the classical stream could be clearer

File: step-framework-parent/step-framework-collections/src/main/java/step/core/accessors/Accessor.java [hunk 2]
```diff
@@ -83,10 +83,18 @@ public interface Accessor<T extends AbstractIdentifiableObject> {
...
+	 * Use streamCloseable instead within a try-with-resources statement for large volume of data
 	 * @return a {@link Stream} with all objects
 	 */
 	Stream<T> stream();
 
+	/**
+	 * Warning: this method must be used within a try-with-resources statement or similar control structure to ensure that the stream's I/O resources are closed promptly after the stream's operations have completed.
+	 * @return a {@link Stream} with all objects
+	 */
+	Stream<T> streamCloseable();
```
Comment (jeromecomte) [inline step-framework-parent/step-framework-collections/src/main/java/step/core/accessors/Accessor.java:96]: Also not sure about the naming but I don't have a better idea. The main difference with the legacy method is that it is performing "real" streaming. Don't know how to reflect this in the name

File: step-framework-parent/step-framework-server/src/main/java/step/framework/server/tables/service/TableService.java [hunk 3]
```diff
@@ -83,7 +85,7 @@ private <T> Stream<T> _request(Collection<T> collection, Table<T> table, Filter
         BiFunction<T, Session<?>, T> enricher = table.getResultItemEnricher().orElse((a, b) -> a);
 
         //return enrich stream
-        return collection.find(filter, searchOrder, request.getSkip(), request.getLimit(), table.getMaxFindDuration().orElse(defaultMaxFindDuration))
+        return collection.findCloseableStream(filter, searchOrder, request.getSkip(), request.getLimit(), table.getMaxFindDuration().orElse(defaultMaxFindDuration))
```
Comment (jeromecomte) [inline step-framework-parent/step-framework-server/src/main/java/step/framework/server/tables/service/TableService.java:88]: We should ensure that the client is closing this resource

**Review by jeromecomte — APPROVED** [review]
Approving with a few minor remarks

File: step-framework-parent/step-framework-collections/src/main/java/step/core/accessors/Accessor.java (hunk 2 above)
Comment (david-stephan) [reply step-framework-parent/step-framework-collections/src/main/java/step/core/accessors/Accessor.java]: renamed to fetchStream

Comment (david-stephan) [reply step-framework-parent/step-framework-collections/src/main/java/step/core/accessors/Accessor.java]: renamed to fetchLazy

**Pull request merged**